<launch>


<group ns="marker_finder">
    <arg name="queue"       default="1000"  />
    <arg name="inTopic"     default="/k01/ir/image_rect" /> <!-- /k01/ir/image_rect  /kinect2/sd/image_ir_rect-->
    <arg name="outTopic"    default="/k01/transform" />

    <node pkg="marker_finder" type="node" name="finder1" output="screen">
        <param name="queue"          value="$(arg queue)"            />
        <param name="outTopic"       value="$(arg outTopic)"        />
        <param name="inTopic"        value="$(arg inTopic)"         />
    </node>

    <node pkg="marker_finder" type="node" name="finder2" output="screen">
        <param name="queue"          value="$(arg queue)"            />
        <param name="outTopic"       value="/k02/transform"        />
        <param name="inTopic"        value="/k02/ir/image_rect"         />
    </node>

    <node pkg="marker_finder" type="node" name="finder3" output="screen">
        <param name="queue"          value="$(arg queue)"            />
        <param name="outTopic"       value="/k03/transform"       />
        <param name="inTopic"        value="/k03/ir/image_rect"         />
    </node>

    <node pkg="marker_finder" type="node" name="finder4" output="screen">
        <param name="queue"          value="$(arg queue)"            />
        <param name="outTopic"       value="/k04/transform"       />
        <param name="inTopic"        value="/k04/ir/image_rect"          />
    </node>

    <node pkg="marker_finder" type="node" name="finder5" output="screen">
        <param name="queue"          value="$(arg queue)"            />
        <param name="outTopic"       value="/k05/transform"       />
        <param name="inTopic"        value="/k05/ir/image_rect"         />
    </node>
</group>

<group ns="calibration">
    <arg name="node_name"                          default="calibration"  />
    <node pkg="wand_calibration" type="node" name="$(arg node_name)" output="screen"> 
        <rosparam command="load" file="$(find wand_calibration)/config/network.yaml" />
        <param name="node_name" type="str" value="$(arg node_name)" />
        <param name="max_pose_stdev" type="double" value="0.005" />
    </node>
</group>


<group ns="play_bag">
   <param name="/use_sim_time" value="true" />

  <arg name="bag_path" default="/home/thomas/Documenti/MasterThesis/bagfiles/calib_ir/filtered4.bag" />

  <!-- Convert compressed images to raw images -->
  <node pkg="image_transport" type="republish" name="republish_k01_depth" args="compressedDepth in:=/k01/depth/image_rect raw out:=/k01/depth/image_rect" />
  <node pkg="image_transport" type="republish" name="republish_k01_ir" args="compressed in:=/k01/ir/image_rect raw out:=/k01/ir/image_rect" />

  <node pkg="image_transport" type="republish" name="republish_k02_depth" args="compressedDepth in:=/k02/depth/image_rect raw out:=/k02/depth/image_rect" />
  <node pkg="image_transport" type="republish" name="republish_k02_ir" args="compressed in:=/k02/ir/image_rect raw out:=/k02/ir/image_rect" />

  <node pkg="image_transport" type="republish" name="republish_k03_depth" args="compressedDepth in:=/k03/depth/image_rect raw out:=/k03/depth/image_rect" />
  <node pkg="image_transport" type="republish" name="republish_k03_ir" args="compressed in:=/k03/ir/image_rect raw out:=/k03/ir/image_rect" />

  <node pkg="image_transport" type="republish" name="republish_k04_depth" args="compressedDepth in:=/k04/depth/image_rect raw out:=/k04/depth/image_rect" />
  <node pkg="image_transport" type="republish" name="republish_k04_ir" args="compressed in:=/k04/ir/image_rect raw out:=/k04/ir/image_rect" />

  <node pkg="image_transport" type="republish" name="republish_k05_depth" args="compressedDepth in:=/k05/depth/image_rect raw out:=/k05/depth/image_rect" />
  <node pkg="image_transport" type="republish" name="republish_k05_ir" args="compressed in:=/k05/ir/image_rect raw out:=/k05/ir/image_rect" />

  <!-- Play bag -->
  <node pkg="rosbag" type="play" name="player" output="screen" required="true" args="--clock --delay 5 --quiet $(arg bag_path)" />

</group>

<!-- <node pkg="rosrun" type="rqt_tf_tree" name="player" output="screen" required="true" args="-clock -delay 5 -quiet $(arg bag_path)" /> -->


</launch>